mm-go-irckit/service.go

```diff
@@ -263,27 +264,37 @@ func scrollback(u *User, toUser *User, args []string, service string) {
 	}
 
 	if len(args) != 2 {
-		u.MsgUser(toUser, "need SCROLLBACK <channel> <lines>")
+		u.MsgUser(toUser, "need SCROLLBACK (#<channel>|@<user>) <lines>")
 		u.MsgUser(toUser, "e.g. SCROLLBACK #bugs 10 (show last 10 lines from #bugs)")
 		return
 	}
 
 	limit, err := strconv.Atoi(args[1])
 	if err != nil {
-		u.MsgUser(toUser, "need SCROLLBACK <channel> <lines>")
+		u.MsgUser(toUser, "need SCROLLBACK (#<channel>|@<user>) <lines>")
 		u.MsgUser(toUser, "e.g. SCROLLBACK #bugs 10 (show last 10 lines from #bugs)")
 		return
 	}
 
-	if !strings.Contains(args[0], "#") {
-		u.MsgUser(toUser, "need SCROLLBACK <channel> <lines>")
+	if !strings.HasPrefix(args[0], "#") && !strings.HasPrefix(args[0], "@") {
+		u.MsgUser(toUser, "need SCROLLBACK (#<channel>|@<user>) <lines>")
 		u.MsgUser(toUser, "e.g. SCROLLBACK #bugs 10 (show last 10 lines from #bugs)")
 		return
 	}
 
-	args[0] = strings.ReplaceAll(args[0], "#", "")
+	var channelName string
+	if strings.Contains(args[0], "@") {
+		args[0] = strings.ReplaceAll(args[0], "@", "")
+		// We need to sort the two user IDs to construct the DM
+		// channel name.
+		userIDs := []string{u.User, u.br.GetUserByUsername(args[0]).User}
```

Comment:
Looking elsewhere, in updatelastviewed(), it calls u.Srv.HasUser(args[0]) and checks for .Ghost. Should I be using u.Srv.HasUser() instead of u.br.GetUserByUsername() here too? Also, what's the .Ghost check for?

Reply:
calling u.Srv.HasUser() is better, as it doesn't need to call out to mattermost to retrieve info it already should have as state.

ghost is everybody but ourselves, doing stuff on our own user (used to?) causes issues sometimes.